src/gui/statsdialog.cpp

```diff
@@ -24,83 +24,81 @@
  * but you are not obligated to do so. If you do not wish to do so, delete this
  * exception statement from your version.
- *
- * Contact : daymansmail@gmail.com
  */
 
 #include "statsdialog.h"
-#include "ui_statsdialog.h"
 
-#include "base/utils/misc.h"
-#include "base/utils/string.h"
+#include "base/bittorrent/cachestatus.h"
 #include "base/bittorrent/session.h"
 #include "base/bittorrent/sessionstatus.h"
-#include "base/bittorrent/cachestatus.h"
 #include "base/bittorrent/torrenthandle.h"
+#include "base/utils/misc.h"
+#include "base/utils/string.h"
+#include "ui_statsdialog.h"
 
 StatsDialog::StatsDialog(QWidget *parent)
     : QDialog(parent)
-    , ui(new Ui::StatsDialog)
+    , m_ui(new Ui::StatsDialog)
 {
-  ui->setupUi(this);
-  setAttribute(Qt::WA_DeleteOnClose);
-  connect(ui->buttonOK, SIGNAL(clicked()), SLOT(close()));
-  updateUI();
-  t = new QTimer(this);
-  t->setInterval(1500);
-  connect(t, SIGNAL(timeout()), SLOT(updateUI()));
-  t->start();
-  show();
+    m_ui->setupUi(this);
+    setAttribute(Qt::WA_DeleteOnClose);
+    connect(m_ui->buttonOK, &QPushButton::clicked, this, &StatsDialog::close);
+
+    update();
+    connect(BitTorrent::Session::instance(), &BitTorrent::Session::statsUpdated
+            , this, &StatsDialog::update);
+    show();
 }
 
-StatsDialog::~StatsDialog() {
-  t->stop();
-  delete t;
-  delete ui;
+StatsDialog::~StatsDialog()
+{
+    delete m_ui;
 }
 
-void StatsDialog::updateUI() {
-  BitTorrent::SessionStatus ss = BitTorrent::Session::instance()->status();
-  BitTorrent::CacheStatus cs = BitTorrent::Session::instance()->cacheStatus();
+void StatsDialog::update()
+{
+    const BitTorrent::SessionStatus &ss = BitTorrent::Session::instance()->status();
+    const BitTorrent::CacheStatus &cs = BitTorrent::Session::instance()->cacheStatus();
 
-  // Alltime DL/UL
-  quint64 atd = BitTorrent::Session::instance()->getAlltimeDL();
-  quint64 atu = BitTorrent::Session::instance()->getAlltimeUL();
-  ui->labelAlltimeDL->setText(Utils::Misc::friendlyUnit(atd));
-  ui->labelAlltimeUL->setText(Utils::Misc::friendlyUnit(atu));
-  // Total waste (this session)
-  ui->labelWaste->setText(Utils::Misc::friendlyUnit(ss.totalWasted()));
-  // Global ratio
-  ui->labelGlobalRatio->setText(
-        ( atd > 0 && atu > 0 ) ?
-          Utils::String::fromDouble((qreal)atu / (qreal)atd, 2) :
-          "-"
-          );
-  // Cache hits
-  qreal readRatio = cs.readRatio();
-  ui->labelCacheHits->setText((readRatio >= 0) ? Utils::String::fromDouble(100 * readRatio, 2) : "-");
-  // Buffers size
-  ui->labelTotalBuf->setText(Utils::Misc::friendlyUnit(cs.totalUsedBuffers() * 16 * 1024));
-  // Disk overload (100%) equivalent
-  // From lt manual: disk_write_queue and disk_read_queue are the number of peers currently waiting on a disk write or disk read
-  // to complete before it receives or sends any more data on the socket. It's a metric of how disk bound you are.
+    // Alltime DL/UL
```

Comment:
I understand that it is quite late for such concerns, but so much of reformatting should be in a commit of its own.